src/http_server.cc

```diff
@@ -2901,6 +2934,18 @@ HTTPAPIServer::HandleInfer(
             infer_request.get());
       }
     }
+    if (err == nullptr && header_forward_pattern_ != "") {
+      HeaderSearchPayload header_search_payload;
+      header_search_payload.error_ = nullptr;
+      header_search_payload.regex_ = header_forward_pattern_.c_str();
```

Comment:
Use constructor for the struct? Then you can have
```
HeaderSearchPayload(const std::string& regex, TritonRequest* request)
```
And the struct can directly store `const std::string& regex_;`